src/dropdown/dropdown.ts

```diff
@@ -312,7 +312,7 @@ export class NgbDropdown implements AfterContentInit, OnDestroy {
 
     // closing on Enter / Space
     if (key === Key.Space || key === Key.Enter) {
-      if (isEventFromItems && (this.autoClose === true || this.autoClose === 'inside')) {
+      if (itemElement && isEventFromItems && (this.autoClose === true || this.autoClose === 'inside')) {
```

Comment:
The check with itemElement and isEventFromItems is redundant, as they are set only if an element is found. isEventFromItems can be removed completely, and the checks can by relied on the existence of itemElement.

Reply:
That's one of the cases, where typesctipt can't make the assumption you've just made

I could change this:
```ts
if (itemElement && isEventFromItems && (this.autoClose === true || this.autoClose === 'inside')) {
  fromEvent(itemElement, 'click').pipe(take(1)).subscribe(() => this.close());
}
```

to this:

```ts
if (isEventFromItems && (this.autoClose === true || this.autoClose === 'inside')) {
  fromEvent(itemElement !, 'click').pipe(take(1)).subscribe(() => this.close());
}
```

WDYT ?

Reply:
I think will change to `// 2` as you're suggesting. Thanks.

Reply:
Yes, I personally prefer 2, as there is only one variable to manage the flag and the element, so no redundancy.